src/kifshare/ui_template.clj

```diff
@@ -1,221 +1,32 @@
 (ns kifshare.ui-template
-  (:use [hiccup.core]
-        [hiccup.page :only [include-css include-js html5]]
-        [kifshare.common :only [layout]])
   (:require [kifshare.config :as cfg]
             [clostache.parser :as prs]
             [clojure.tools.logging :as log]
             [cheshire.core :as json])
   (:import [org.apache.commons.io FileUtils]))
 
-(defn clear
+(defn read-template
   []
-  (log/debug "entered kifshare.ui-template/clear")
-  (html [:div {:class "clear"}]))
+  (slurp "resources/ui.xml"))
 
-(defn section-spacer [] (html [:div {:class "section-spacer"}]))
-
-(defn irods-avu-row
-  [mmap]
-  (log/debug "entered kifshare.ui-template/irods-avu-row")
-
-  (html
-   [:tr
-    [:td {:title (:attr mmap)} (:attr mmap)]
-    [:td {:title (:value mmap)} (:value mmap)]
-    [:td {:title (:unit mmap)} (:unit mmap)]]))
-
-(defn irods-avu-table
-  [metadata]
-  (log/debug "entered kifshare.ui-template/irods-avu-table")
-
-  (if (pos? (count metadata))
-    (html
-     [:div {:id "irods-avus"}
-      [:div {:id "irods-avus-header"}
-       [:h2 "Metadata"]]
-      [:table {:id "irods-avus-data"}
-       [:thead
-        [:tr
-         [:th "Attribute"]
-         [:th "Value"]
-         [:th "Unit"]]]
-       [:tbody
-        (map irods-avu-row metadata)]]
-      (section-spacer)])))
-
-(defn lastmod
-  [ticket-info]
-  (log/debug "entered kifshare.ui-template/lastmod")
-
-  (html
-   [:div {:id "lastmod-detail"}
-    [:div {:id "lastmod-label"}
-     [:p "Last Modified:"]]
-    [:div {:id "lastmod"}
-     [:p (:lastmod ticket-info)]]]))
-
-(defn filesize
-  [ticket-info]
-  (log/debug "entered kifshare.ui-template/filesize")
-
-  (html
-   [:div {:id "size-detail"}
-    [:div {:id "size-label"}
-     [:p "File Size:"]]
-    [:div {:id "size"}
-     [:p (FileUtils/byteCountToDisplaySize
-          (Long/parseLong (:filesize ticket-info)))]]]))
+(def memo-read-template
+  (memoize read-template))
```

Comment:
`read-template` isn't getting called anywhere else, so I would push `memoize` down into the `read-template` function and get rid of `memo-read-template`.